src/abstractblocksparsearray/abstractblocksparsematrix.jl

```diff
@@ -1 +1,22 @@
 const AbstractBlockSparseMatrix{T} = AbstractBlockSparseArray{T,2}
+
+# SVD is implemented by trying to
+# 1. Attempt to find a block-diagonal implementation by permuting
+# 2. Fallback to AbstractBlockArray implementation via BlockedArray
+function svd(
```

Comment:
So I see this is designed around a private `BlockSparseArrays.svd` function, is the plan to overload `LinearAlgebra.svd`? Probably we should define this as:
```julia
@interface ::AbstractBlockSparseArrayInterface function svd(...)
```
(I would like to make the `AbstractArrayInterface` interface types take an `ndims` parameter so it could be `AbstractBlockSparseMatrixInterface` but that isn't supported yet) and then derive that function for `LinearAlgebra.svd(::AnyAbstractBlockSparseMatrix)` in the style of `Derive.jl`.